### ISSUE TITLE
Collation case level examples debug

### ISSUE THREAD

**Issue by echeran** [unicode-org/icu4x#2472]
When creating docs examples for the case level setting in collation options for collator (#2447), I couldn't seem to create examples that returned distinct outputs from primary strength versus secondary strength.

The examples in #2447 are adapted from the examples provided in the [ICU User Guide for case level](https://unicode-org.github.io/icu/userguide/collation/concepts.html#caselevel), partly due to not being able to get the code to produce the output expected based on the User Guide.

Is the application of case-level in collation broken?  Or is it a problem of not knowing how to use it?

**Comment by markusicu** [comment]
I suggest you try this list of strings and compare each with the next:
- ⓓⓔⓐⓛ
- DEAL
- dejavu
- dejAvu
- déjavu

The last two trade places in the sort order depending on the settings:
- primary+case ignores the accent difference so dejAvu will compare greatest
- secondary+case has the accent difference win over the case difference

----
This is a bug:
```Rust
assert_eq!(primary_and_case.compare("DEAL", "ⓓⓔⓐⓛ"), Ordering::Equal);
```
This should yield Greater.

**Comment by echeran** [comment]
@markusicu - Thanks! I incorporated some of that into PR #2447.

Note to self/others: Leave this issue open for the bug identified in the comment above.

**Comment by markusicu** [comment]
> @markusicu - Thanks! I incorporated some of that into PR #2447.

Nice.

> Note to self/others: Leave this issue open for the bug identified in the comment above.

That is now visible on two lines in the above PR:
```
//! assert_eq!(primary_and_case.compare("ⓓⓔⓐⓛ", "DEAL"), Ordering::Equal);
...
//! assert_eq!(primary_and_case.compare("déjavu", "dejAvu"), Ordering::Equal);```

**Comment by echeran** [comment]
This issue was reported again in #2884 , which was fixed by #2892. Closing this as duplicated and fixed.
